src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -97,7 +116,56 @@ def __call__(
             else:
                 parsed_args.append(arg)
 
-        result = self.func(session, *parsed_args)
+                # Initialize import directory
+
+        temporary_import_path = tempfile.TemporaryDirectory()
+        last_import_directory = sys._xoptions.get("snowflake_import_directory")
+        sys._xoptions["snowflake_import_directory"] = temporary_import_path.name
+
+        # Save a copy of module cache
+        frozen_sys_module_keys = set(sys.modules.keys())
+
+        def cleanup_imports():
+            for module_path in self.imports:
+                if module_path in sys.path:
+                    sys.path.remove(module_path)
```

Comment:
I see in a previous [PR](https://github.com/snowflakedb/snowpark-python/pull/1335/files) we track udf imports inside a session
`analyzer.session.udf._udf_import_directories[udf_name]`.

so it doesn't apply here?

Reply:
I've removed the logic around import directories in both UDF and stored proc implementation, there's no need to track them.